sonic_platform_base/sonic_xcvr/api/public/c_cmis.py

```diff
@@ -150,6 +152,46 @@ def set_tx_power(self, tx_power):
         time.sleep(1)
         return status
 
+    def util_status_check_with_retry(self, retry, sleep_time, reg_field, expected_status):
```

Comment:
@jaganbal we don't need retry. Let the API return failure after timeout

<img width="783" alt="image" src="https://github.com/sonic-net/sonic-platform-common/assets/45705344/a387f8a9-1003-4e6a-a014-2d66c7da01f9">